tutorials/nservicebus-101/lesson-2/tutorial.md

```diff
@@ -0,0 +1,191 @@
+
+When sending a message, the endpoint's message serializer will serialize an instance of the `DoSomething` class (let's say to JSON) and add that to the contents of the outgoing message that goes to the queue. On the other end, the receiving endpoint will deserialize the message from JSON back to an instance of the message class so that it can be used in code.
+
+Messages can support whatever data the serializer can successfully process. Thus messages can even contain child objects or collections.
+
+snippet:ComplexCommand
+
+While it can be pretty tempting to throw in the kitchen sink, consider that messages are a contract between two endpoints. Any change to the message will likely involve a change on both the sender and receiver side. The more properties you have on a message, the more reasons it has to change, so keep your messages as slim as possible.
+
+Also, you should not embed logic within your NServiceBus message classes. Each message should contain only automatic properties and not computed properties or methods. Also, it is a good practice to instantiate collection properties from a default parameterless constructor as shown above, so that you never have to deal with a potentially null collection.
+
+In essence, messages should be carriers for data only. This way, discovering a bug in logic will not require redeployment of every endpoint that uses a message.
+
+
+## How do I organize messages?
+
+Because messages must be shared between multiple endpoints, you can't put the classes in the same assembly with the endpoints. They need to live in a separate class library so that they can be referenced by multiple endpoints.
+
+A **message assembly** should contain only NServiceBus message contracts, and any supporting types required by the messages themselves. For example, a message may use an enumeration type for one of its properties; that enumeration should also live within the same message assembly.
+
+Message assemblies should be entirely self-contained. They should have no dependencies other than libraries included with the .NET Framework, and the NServiceBus core assembly, which is required to reference the `ICommand` interface. Limiting dependencies makes your message contracts more resilient to future changes in the system.
+
+NOTE: In another course, we'll learn how (and why) to use [message conventions](/nservicebus/messaging/conventions.md) to identify message types, rather than the `ICommand` interface. Then we don't even need to reference the NServiceBus assembly in our message assemblies. But, we'll leave that out of scope for now.
```

Comment:
"message conventions" are not "real life". they solve a specific problem that very few organisation have.